docs/uptime-guide/deployment-arch.asciidoc

```diff
@@ -0,0 +1,13 @@
+[role="xpack"]
+[[uptime-deployment-arch]]
+== Deployment Architecture
+
+There are infinite ways to deploy Uptime and Heartbeat, some of which work better than others. A guiding principle in these recommendations is that an outage that takes down the service being monitored should not also take down Heartbeat. You will want Heartbeat to be functioning even when your site is not, so the guidelines that follow attempt to minimize this possibility.
+
+Heartbeat is generally run as a centralized service within a datacenter. While it is possible to run it as a separate "sidecar" process paired with each process/container we recommend against it. Running Heartbeat centrally ensures you will still be able to see monitoring data in the event of an overloaded, disconnected, or otherwise malfunctioning server. 
+
+For further redundancy, you may want to deploy multiple Heartbeats across geographic and/or network boundaries to provide more data. Specify Heartbeat's observer https://www.elastic.co/guide/en/beats/heartbeat/current/configuration-observer-options.html[geo options] to do so. Some examples might be:
```

Comment:
I've seen us use `{beatname} agent` and just plain `{beatname}` in various places. Heartbeat isn't really an agent (an agent implies it being a sidecar, which heartbeat is not).

My sense here is that colloquially the `agent/instance` clarification is generally implied by context and omitted for brevity, and only explicitly added for additional emphasis and disambiguation where required.

Would love to hear from @dedemorton for some guidance here.

Reply:
@andrewvc We shouldn't say Heartbeats (plural). Most style guides advise against making product names plural because it impacts branding and trademarking. 

My recommendation is:

- When you only have one instance of a Beat running, there's no need to qualify it.

- If you are talking about multiple instances, you should say "Heartbeat instances" rather than "Heartbeats".

Also, you should use the `{heartbeat}` shared attribute to resolve the Beat name. This makes it easier for us to change the name if we ever have to go through rebranding. Using the attribute also safeguards against weird typos, like Heatbeat. hehe.